Mail/Views/Thread List/ThreadListable.swift

```diff
@@ -0,0 +1,35 @@
+/*
+ Infomaniak Mail - iOS App
+ Copyright (C) 2024 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+import MailCore
+
+enum ThreadListableType {
+    case search
+    case threadList
+}
+
+@MainActor protocol ThreadListable {
+    var mailboxManager: MailboxManager { get }
+    var frozenFolder: Folder { get }
+    var origin: ThreadListableType { get }
```

Comment:
`origin` seems to be only used for matomo.
Maybe we can directly have a property `matomoEventCategory: MatomoUtils.EventCategory` declared in each model (and conforming to the protocol)

Reply:
Oops missed it. Maybe then we can use casting instead of needing an enum.
Something like `viewModel is SearchViewModel ? true : false` ?